### ISSUE TITLE
Eventflow Resolves Services on Startup Without Disposing?

### ISSUE THREAD

**Issue by ylbillyli** [eventflow/EventFlow#340]
Hi @rasmus,

I am using EventFlow with Autofac. Recently I notice on startup, I have a few resources that get on hold never released. Tracing it down, looks like it is Eventflow's `CreateContainer` ->  `AutofacServiceRegistration.CreateResolver` -> `autofacRootResolver.ValidateRegistrations()` that resolve all my services (which some are `IDisposable` e.g. `IMyDbContext`) without disposing them. I end up having db connection opened for the lifetime of my app. Is this design issue on my end or should EventFlow wrap the check in a `LifetimeScope` like this?
```
public IRootResolver CreateResolver(bool validateRegistrations)
{
	var container = _containerBuilder.Build();
	var autofacRootResolver = new AutofacRootResolver(container);

	if (validateRegistrations)
	{
		using (var scopedResolver = container.BeginLifetimeScope("validateRegistrations"))
	        {
		        scopedResolver.ValidateRegistrations();
	        }
	}

	return new AutofacRootResolver(container);
}
```  
  
I am forced to not do validation at the moment to get around `EventFlowOptions.New....CreateContainer(false)`

**Comment by rasmus** [comment]
@ylbillyli Yes, that does look like a bug as all the resources are resolved using the main container which is never disposed. I'll create a fix for that

**Comment by rasmus** [comment]
@ylbillyli fixed in v0.46.2886

**Comment by ylbillyli** [comment]
wow, that's quick, thx @rasmus 👍
